.travis.yml

```diff
@@ -1,20 +1,54 @@
+os: linux
+dist: trusty
+sudo: false
+group: edge
+
 language: python
 python:
-  - "2.6"
-  - "2.7"
-  - "3.3"
-  - "3.4"
-  - "3.5"
-  - "3.6"
+- 2.6
+- 2.7
+- 3.3
+- 3.4
+- 3.5
+- &mainstream_python 3.6
+- nightly
 
 install:
-  - "travis_retry pip install nose"
-  - "travis_retry pip install coveralls"
-  - "travis_retry pip install webassets"
-  - "travis_retry pip install mock"
-  - "travis_retry pip install --pre -e ."
+- travis_retry pip install --pre -e .[testing,webassets]
 
-script: "nosetests -v --with-coverage --cover-package=dukpy --cover-erase"
+script:
+- nosetests -v --with-coverage --cover-package=dukpy --cover-erase
+- if python -c "import sys; sys.exit(sys.version_info < (2,7))"; then
+      python setup.py check -rm;
+  fi
+- if python -c "import sys; sys.exit(sys.version_info < (3,6))"; then
+      python setup.py check -s;
+  fi
 
 after_success:
-  coveralls
+- coveralls
+
+stages:
+- test
+- name: deploy
+  if: tag IS present
+
+jobs:
+  fast_finish: true
+  allow_failures:
+  - python: nightly
+  include:
+  - stage: deploy
+    python: *mainstream_python
+    install: skip
+    script: skip
+    deploy:
+      provider: pypi
+      on:
+        tags: true
+        all_branches: true
```

Comment:
Maybe we want this only on master if it does what I expect?

Reply:
It would only affect tagged commits, I don't see reason to remove it. But it shouldn't change things a lot if missing. Feel free to delete it if you want.